src/hfdl.rs

```diff
@@ -0,0 +1,629 @@
+            LPDUSrcType::Aircraft => serializer.serialize_str("Aircraft"),
+            LPDUSrcType::GroundStation => serializer.serialize_str("Ground station"),
+        }
+    }
+}
+
+impl TryFrom<String> for LPDUSrcType {
+    type Error = String;
+
+    fn try_from(value: String) -> Result<Self, Self::Error> {
+        match value.as_str() {
+            "Aircraft" => Ok(LPDUSrcType::Aircraft),
+            "Ground station" => Ok(LPDUSrcType::GroundStation),
+            _ => Err(format!("Unknown LPDUSrcType: {}", value))
+        }
+    }
+}
+
+#[derive(Serialize, Deserialize, Debug, Clone, Default)]
+#[serde(deny_unknown_fields)]
+pub struct LPDUType {
+    name: String,
+    id: u16,
+}
+
+#[derive(Serialize, Deserialize, Debug, Clone, Default)]
+#[serde(deny_unknown_fields)]
+pub struct LPDUAircraftInfo {
+    icao: String
+}
+
+#[derive(Serialize, Deserialize, Debug, Clone, Default)]
+#[serde(deny_unknown_fields)]
+pub struct LPDUHfnPdu {
+    err: bool,
+    #[serde(rename = "type")]
+    lpdu_type: LPDUType,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    flight_id: Option<String>,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pos: Option<Position>,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    utc_time: Option<UTCTime>,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    freq_data: Option<Vec<LPDUFreqData>>,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    version: Option<u8>,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    time: Option<UTCTime>,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    flight_leg_num: Option<u16>,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    gs: Option<SPDUorLPDUSource>,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    frequency: Option<FreqId>,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    freq_search_cnt: Option<LPDUHfnPduCount>,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    hfdl_disabled_duration: Option<LPDUHfnPduDisabledCount>,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pdu_stats: Option<PDUStats>,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    last_freq_change_cause: Option<LastFreqChangeCause>,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    acars: Option<LPDUAcars>,
+}
+
+#[derive(Serialize, Deserialize, Debug, Clone, Default)]
+#[serde(deny_unknown_fields)]
+pub struct LastFreqChangeCause {
+    code: u8,
+    descr: String,
+}
+
+#[derive(Serialize, Deserialize, Debug, Clone, Default)]
+#[serde(deny_unknown_fields)]
+pub struct PDUStats {
+    mpdus_rx_ok_cnt: PDUStatCounts,
+    mpdus_rx_err_cnt: PDUStatCounts,
+    mpdus_tx_cnt: PDUStatCounts,
+    mpdus_delivered_cnt: PDUStatCounts,
+    spdus_rx_ok_cnt: u16,
+    spdus_missed_cnt: u16,
+}
+
+#[derive(Serialize, Deserialize, Debug, Clone, Default)]
+#[serde(deny_unknown_fields)]
+pub struct PDUStatCounts {
+    #[serde(rename = "300bps")]
+    three_hunderd_bps: u8,
```

Comment:
Found the copy/paste typo, hunderd -> hundred

Reply:
One typo and copilot running with it ;)